ethash/src/compute.rs

```diff
@@ -238,25 +242,34 @@ fn get_data_size(block_number: u64) -> usize {
 	sz as usize
 }
 
-
 /// Difficulty quick check for POW preverification
 ///
 /// `header_hash`      The hash of the header
 /// `nonce`            The block's nonce
 /// `mix_hash`         The mix digest hash
 /// Boundary recovered from mix hash
 pub fn quick_get_difficulty(header_hash: &H256, nonce: u64, mix_hash: &H256) -> H256 {
-	let mut buf = [0u8; 64 + 32];
-	unsafe { ptr::copy_nonoverlapping(header_hash.as_ptr(), buf.as_mut_ptr(), 32) };
-	unsafe { ptr::copy_nonoverlapping(mem::transmute(&nonce), buf[32..].as_mut_ptr(), 8) };
+	unsafe {
+		// This is safe - the `sha3_512` call below reads the first 40 bytes (which we explicitly set
+		// with two `copy_nonoverlapping` calls) but writes the first 64, and then we explicitly write
+		// the next 32 bytes before we read the whole thing with `sha3_256`.
+		//
+		// This cannot be elided by the compiler as it doesn't know the implementation of
+		// `sha3_512`.
+		let mut buf: [u8; 64 + 32] = mem::uninitialized();
+
+		ptr::copy_nonoverlapping(header_hash.as_ptr(), buf.as_mut_ptr(), 32);
```

Comment:
copy_from_slice is exactly the same thing as copy_nonoverlapping [link](https://doc.rust-lang.org/src/core/slice/mod.rs.html#667)